service/controller/app/v1/resource/chart/desired.go

```diff
@@ -44,24 +45,40 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 			Annotations: cr.GetAnnotations(),
 		},
 		Spec: v1alpha1.ChartSpec{
+			Config:     config,
 			Name:       cr.GetName(),
 			Namespace:  key.Namespace(cr),
 			TarballURL: tarballURL,
 		},
 	}
 
-	if key.AppConfigMapName(cr) != "" || appcatalogkey.ConfigMapName(cc.AppCatalog) != "" {
-		config := v1alpha1.ChartSpecConfig{
-			ConfigMap: v1alpha1.ChartSpecConfigConfigMap{
+	return chartCR, nil
+}
+
+func generateConfig(cr v1alpha1.App, appCatalog v1alpha1.AppCatalog) v1alpha1.ChartSpecConfig {
+	config := v1alpha1.ChartSpecConfig{}
+
+	if hasConfigMap(cr, appCatalog) || hasSecret(cr, appCatalog) {
```

Comment:
That condition is not necessary. The specific conditions are below and do what they should.

Reply:
Thanks. I've removed it.